<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>GeneticData • compbio2019</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="GeneticData">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio2019</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/00_Syllabus_and_Expectations.html">Syllabus</a>
    </li>
    <li>
      <a href="../articles/01_Getting_Started_with_R.html">Starting with R</a>
    </li>
    <li>
      <a href="../articles/02_Starting_with_Data.html">Starting with data</a>
    </li>
    <li>
      <a href="../articles/03_Manipulating_Data.html">Manipulating, analyzing and exporting data with tidyverse</a>
    </li>
    <li>
      <a href="../articles/04-plotting.html">Data visualization with ggplot2</a>
    </li>
    <li>
      <a href="../articles/05-Functions.html">Functions and repeatability</a>
    </li>
    <li>
      <a href="../articles/06_Exploration_Setup.html">Exploration</a>
    </li>
    <li>
      <a href="../articles/07_Exploration_Hands_On.html">Exploration Hands-On</a>
    </li>
    <li>
      <a href="../articles/08_Tree_of_life.html">GeneticData</a>
    </li>
    <li>
      <a href="../articles/09_GBIF_and_Location.html">GBIF And Spatial Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>GeneticData</h1>
            
      
      
      <div class="hidden name"><code>08_Tree_of_life.Rmd</code></div>

    </div>

    
    
<div id="the-tree-of-life" class="section level2">
<h2 class="hasAnchor">
<a href="#the-tree-of-life" class="anchor"></a>The Tree of Life</h2>
<p>First, we need to do some installs.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"devtools"</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/remote-reexports">install_github</a></span>(<span class="st">"ropensci/rotl"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/remote-reexports">install_github</a></span>(<span class="st">"ropensci/traits"</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">'phylotaR'</span>)</a></code></pre></div>
<p>Now, load your packages:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(rotl)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(phylotaR)</a></code></pre></div>
</div>
<div id="open-tree" class="section level2">
<h2 class="hasAnchor">
<a href="#open-tree" class="anchor"></a>Open Tree</h2>
<p>First, we’re going to look for some data. Discuss with your neighbor a clade you like and want to build a tree of. For example, perhaps I really like hominids.</p>
<pre><code>##   search_string unique_name approximate_match ott_id is_synonym flags
## 1         pongo       Pongo             FALSE 417949      FALSE      
## 2           pan         Pan             FALSE 417957      FALSE      
## 3       gorilla     Gorilla             FALSE 417969      FALSE      
## 4       hoolock     Hoolock             FALSE 712902      FALSE      
## 5          homo        Homo             FALSE 770309      FALSE      
##   number_matches
## 1              2
## 2              2
## 3              1
## 4              1
## 5              1</code></pre>
<p>Put the names of five genera into the function. If you want, you can use mine below.</p>
<pre><code>##   search_string unique_name approximate_match  ott_id is_synonym flags
## 1     martialis   Martialis             FALSE  334634      FALSE      
## 2          atta        Atta             FALSE  709964      FALSE      
## 3     ectatomma   Ectatomma             FALSE  162845      FALSE      
## 4     tatuidris   Tatuidris             FALSE  150177      FALSE      
## 5     aneuretus   Aneuretus             FALSE  425343      FALSE      
## 6        xymmer      Xymmer             FALSE 4583488      FALSE      
## 7      aenictus    Aenictus             FALSE  169919      FALSE      
##   number_matches
## 1              1
## 2              1
## 3              1
## 4              1
## 5              1
## 6              1
## 7              1</code></pre>
<p>Try to view them on a tree:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">tr &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/rotl/topics/tol_induced_subtree">tol_induced_subtree</a></span>(<span class="dt">ott_ids =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/rotl/topics/taxonomy-methods">ott_id</a></span>(resolved_names))</a></code></pre></div>
<pre><code>## 
Progress [----------------------------------] 0/43 (  0%) ?s
Progress [================================] 43/43 (100%)  0s
                                                            </code></pre>
<pre><code>## Warning in collapse_singles(tr, show_progress): Dropping singleton
## nodes with labels: mrcaott1682ott35311, mrcaott1682ott34821,
## mrcaott34821ott950983, Aneuretinae ott393261, mrcaott4706ott5525,
## mrcaott5525ott8038, mrcaott8038ott8685, mrcaott8685ott36581,
## mrcaott8685ott56535, mrcaott8685ott72729, mrcaott8685ott86888,
## mrcaott86888ott129302, Attini ott344333, mrcaott114767ott114779,
## mrcaott114779ott339738, mrcaott114779ott116619, mrcaott116619ott148996,
## mrcaott148996ott1071522, mrcaott148996ott150184, mrcaott148996ott339763,
## mrcaott148996ott409299, mrcaott48019ott671955, Ectatomminae ott327970,
## mrcaott48019ott162057, Dorylinae ott161685, mrcaott7438ott7439,
## mrcaott7439ott165217, mrcaott7439ott15430, mrcaott15430ott120483,
## mrcaott15430ott150187, mrcaott15430ott169913, mrcaott17752ott97752,
## mrcaott17752ott150176, mrcaott150176ott1037289, mrcaott150176ott4539262,
## Agroecomyrmecinae ott504610, mrcaott97755ott149123, mrcaott149123ott452719,
## mrcaott149123ott739132, mrcaott149123ott503313, mrcaott503313ott928530,
## mrcaott928530ott6280946, Martialinae ott334635</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tr)</a></code></pre></div>
<p><img src="08_Tree_of_life_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>I’m a little surprised by this tree.</p>
</div>
<div id="looking-for-data" class="section level2">
<h2 class="hasAnchor">
<a href="#looking-for-data" class="anchor"></a>Looking for data</h2>
<p>We’re going to try to build a tree of these taxa from GenBank Data. Much like accessing NOAA data, we will need to make an API Key. Go <a href="https://www.ncbi.nlm.nih.gov/account/">here</a> and register.</p>
<p>Now that we have that key, we can access NCBI:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(rentrez)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_search">entrez_search</a></span>(<span class="dt">db =</span> <span class="st">"Nucleotide"</span>, <span class="dt">term=</span><span class="st">"Ectatomma"</span>, <span class="dt">api_key =</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span>)</a></code></pre></div>
<pre><code>## Entrez search result with 647 hits (object contains 20 IDs and no web_history object)
##  Search term (as translated):  "Ectatomma"[Organism] OR Ectatomma[All Fields]</code></pre>
<p>Holy moley that’s a lot of data. If we do that for every taxon, we’ll end up with a mess. Why don’t we try to filter it a bit.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">Ectatomma &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_search">entrez_search</a></span>(<span class="dt">db =</span> <span class="st">"Nucleotide"</span>, <span class="dt">term=</span><span class="st">"Ectatomma AND 2018:2019[PDAT]"</span>, <span class="dt">retmax =</span> <span class="dv">100</span>, <span class="dt">api_key =</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span>)</a></code></pre></div>
<p>So there is a good amount of data for our Ectatomma ants.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="cf">for</span>( seq_start <span class="cf">in</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq</a></span>(<span class="dv">1</span>,<span class="dv">50</span>,<span class="dv">100</span>)){</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">     recs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db=</span><span class="st">"nuccore"</span>, <span class="dt">id =</span> Ectatomma<span class="op">$</span>ids,</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">                          <span class="dt">rettype=</span><span class="st">"fasta"</span>, <span class="dt">retmax=</span><span class="dv">50</span>, <span class="dt">retstart=</span>seq_start)</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">     <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cat">cat</a></span>(recs, <span class="dt">file=</span><span class="st">"Ect.seqs.fasta"</span>, <span class="dt">append=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">     <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cat">cat</a></span>(seq_start<span class="op">+</span><span class="dv">49</span>, <span class="st">"sequences downloaded</span><span class="ch">\r</span><span class="st">"</span>)</a>
<a class="sourceLine" id="cb12-6" data-line-number="6"> }</a></code></pre></div>
<pre><code>## 50 sequences downloaded</code></pre>
<p>Now, let’s look at this file. What do we notice about it?</p>
<p>All right, let’s move on to trying to write a function that will return all IDs for all of the ants.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">search_year &lt;-<span class="st"> </span><span class="cf">function</span>(ant){</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">    query &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(ant, <span class="st">" AND "</span>, <span class="st">"2018:2019[PDAT]"</span>)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">    search_returns &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_search">entrez_search</a></span>(<span class="dt">db=</span><span class="st">"Nucleotide"</span>, <span class="dt">term=</span>query, <span class="dt">retmax=</span><span class="dv">20</span>,<span class="dt">api_key =</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span>))</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/function">return</a></span>(search_returns)</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">}</a>
<a class="sourceLine" id="cb14-6" data-line-number="6"></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"></a>
<a class="sourceLine" id="cb14-8" data-line-number="8">ants &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(resolved_names<span class="op">$</span>unique_name)</a>
<a class="sourceLine" id="cb14-9" data-line-number="9"></a>
<a class="sourceLine" id="cb14-10" data-line-number="10">labels &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">sapply</a></span>(ants,search_year,  <span class="dt">USE.NAMES =</span> T)</a></code></pre></div>
<p>Have a look at ?sapply. Talk to a partner and see if you can figure out what sapply does.</p>
<p>So what is the problem with this output? It’s hard to parse - let’s make it easier to pull data for one taxon at a time.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">search_year &lt;-<span class="st"> </span><span class="cf">function</span>(ant){</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">    search_returns =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>()</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">    query &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(ant, <span class="st">" AND "</span>, <span class="st">"2018:2019[PDAT]"</span>)</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">    search_returns[ant] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_search">entrez_search</a></span>(<span class="dt">db=</span><span class="st">"Nucleotide"</span>, <span class="dt">term=</span>query, <span class="dt">retmax=</span><span class="dv">20</span>,<span class="dt">api_key =</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span>))</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/function">return</a></span>(search_returns)</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">}</a>
<a class="sourceLine" id="cb15-7" data-line-number="7"></a>
<a class="sourceLine" id="cb15-8" data-line-number="8"></a>
<a class="sourceLine" id="cb15-9" data-line-number="9">ants &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(resolved_names<span class="op">$</span>unique_name)</a>
<a class="sourceLine" id="cb15-10" data-line-number="10"></a>
<a class="sourceLine" id="cb15-11" data-line-number="11">labels &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">sapply</a></span>(ants,search_year,  <span class="dt">USE.NAMES =</span> T)</a></code></pre></div>
<p>There’s one remaining problem, though. Have a look at your outputs. How do you think we can fix this one? Let’s have a look at when papers related to Xymmer were published:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">xymmer_papers &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_search">entrez_search</a></span>(<span class="dt">db=</span><span class="st">"pmc"</span>, <span class="dt">term=</span><span class="st">"Xymmer"</span>, <span class="dt">retmax=</span><span class="dv">20</span>, <span class="dt">rettype=</span> <span class="st">"text"</span>, <span class="dt">api_key =</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">xymmer_papers &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db=</span><span class="st">"pmc"</span>, <span class="dt">term=</span><span class="st">"Xymmer"</span>, <span class="dt">id =</span> xymmer_papers<span class="op">$</span>ids, <span class="dt">retmax=</span><span class="dv">20</span>, <span class="dt">rettype=</span> <span class="st">"text"</span>, <span class="dt">api_key =</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span>)</a></code></pre></div>
<p>Does this give you a hint about something that needs to change to get more data?</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">search_year &lt;-<span class="st"> </span><span class="cf">function</span>(ant){</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">    search_returns =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>()</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">    query &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(ant, <span class="st">" AND "</span>, <span class="st">"2018:2019[PDAT]"</span>)</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">    search_returns[ant] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_search">entrez_search</a></span>(<span class="dt">db=</span><span class="st">"Nucleotide"</span>, <span class="dt">term=</span>query, <span class="dt">retmax=</span><span class="dv">20</span>,<span class="dt">api_key =</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span>))</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/function">return</a></span>(search_returns)</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">}</a>
<a class="sourceLine" id="cb17-7" data-line-number="7"></a>
<a class="sourceLine" id="cb17-8" data-line-number="8"></a>
<a class="sourceLine" id="cb17-9" data-line-number="9">ants &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/vector">as.vector</a></span>(resolved_names<span class="op">$</span>unique_name)</a>
<a class="sourceLine" id="cb17-10" data-line-number="10"></a>
<a class="sourceLine" id="cb17-11" data-line-number="11">labels &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">sapply</a></span>(ants,search_year,  <span class="dt">USE.NAMES =</span> T)</a></code></pre></div>
<p>Now, we’re going to get the genetc data.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">search_seq &lt;-<span class="st"> </span><span class="cf">function</span>(ant){</a>
<a class="sourceLine" id="cb18-2" data-line-number="2">     recs[ant] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db=</span><span class="st">"nuccore"</span>, <span class="dt">id =</span> ant,</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">                          <span class="dt">rettype=</span><span class="st">"fasta"</span>, <span class="dt">retmax=</span><span class="dv">10</span>)</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">     <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cat">cat</a></span>(recs, <span class="dt">file=</span><span class="st">"Ants.seqs.fasta"</span>, <span class="dt">append=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb18-5" data-line-number="5"></a>
<a class="sourceLine" id="cb18-6" data-line-number="6">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/function">return</a></span>(recs)</a>
<a class="sourceLine" id="cb18-7" data-line-number="7">}</a>
<a class="sourceLine" id="cb18-8" data-line-number="8"></a>
<a class="sourceLine" id="cb18-9" data-line-number="9"></a>
<a class="sourceLine" id="cb18-10" data-line-number="10">seq_returns &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">sapply</a></span>(labels, search_seq,  <span class="dt">USE.NAMES =</span> T)</a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#the-tree-of-life">The Tree of Life</a></li>
      <li><a href="#open-tree">Open Tree</a></li>
      <li><a href="#looking-for-data">Looking for data</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
